Linux compile errors

Seems a few errors have creeped in, which g++ is complaining about.

A bunch related to `find`:
```
src/Map/TileMap.cpp: In member function ‘void TileMap::removeMineLocation(const NAS2D::Point_2d&)’:
src/Map/TileMap.cpp:87:76: error: no matching function for call to ‘find(std::vector<NAS2D::Point_2d>::iterator, std::vector<NAS2D::Point_2d>::iterator, const NAS2D::Point_2d&)’
  mMineLocations.erase(find(mMineLocations.begin(), mMineLocations.end(), pt));
                                                                            ^
```

```
src/Things/Structures/RobotCommand.cpp: In member function ‘bool RobotCommand::commandedByThis(Robot*) const’:
src/Things/Structures/RobotCommand.cpp:22:54: error: no matching function for call to ‘find(std::vector<Robot*>::const_iterator, std::vector<Robot*>::const_iterator, Robot*&)’
  return find(mRobotList.begin(), mRobotList.end(), _r) != mRobotList.end();
                                                      ^
```

```
src/Things/Structures/RobotCommand.cpp: In member function ‘void RobotCommand::addRobot(Robot*)’:
src/Things/Structures/RobotCommand.cpp:41:58: error: no matching function for call to ‘find(std::vector<Robot*>::iterator, std::vector<Robot*>::iterator, Robot*&)’
  auto _it = find(mRobotList.begin(), mRobotList.end(), _r);
                                                          ^
```

```
src/Things/Structures/RobotCommand.cpp: In member function ‘void RobotCommand::removeRobot(Robot*)’:
src/Things/Structures/RobotCommand.cpp:66:58: error: no matching function for call to ‘find(std::vector<Robot*>::iterator, std::vector<Robot*>::iterator, Robot*&)’
  auto _it = find(mRobotList.begin(), mRobotList.end(), _r);
                                                          ^
```

```
src/RobotPool.cpp: In member function ‘void RobotPool::erase(Robot*)’:
src/RobotPool.cpp:47:55: error: no matching function for call to ‘find(std::vector<Robot*>::iterator, std::vector<Robot*>::iterator, Robot*&)’
  mRobots.erase(find(mRobots.begin(), mRobots.end(), _r));
                                                       ^
```

```
src/RobotPool.cpp:49:25:   required from here
src/RobotPoolHelper.h:59:16: error: no matching function for call to ‘find(std::vector<Robodigger*>::iterator, std::vector<Robodigger*>::iterator, Robot*&)’
  auto it = find(t.begin(), t.end(), _r);
                ^
```

```
src/RobotPoolHelper.h: In instantiation of ‘void eraseRobot(T&, Robot*) [with T = std::vector<Robodozer*>]’:
src/RobotPool.cpp:50:24:   required from here
src/RobotPoolHelper.h:59:16: error: no matching function for call to ‘find(std::vector<Robodozer*>::iterator, std::vector<Robodozer*>::iterator, Robot*&)’
```

An extra prefix in a header file (probably pasted from the .cpp definition):
```
src/States/GameState.h: At global scope:
src/States/GameState.h:115:7: error: extra qualification ‘GameState::’ on member ‘pullRobotFromFactory’ [-fpermissive]
  void GameState::pullRobotFromFactory(ProductType pt, Factory& factory);
       ^
```

Warning (order of member initializer list doesn't match order of member variables in class):
```
src/Things/Structures/../../ProductionCost.h: In constructor ‘ProductionCost::ProductionCost(int, int, int, int, int)’:
src/Things/Structures/../../ProductionCost.h:44:25: warning: ‘ProductionCost::mTurnsToBuild’ will be initialized after [-Wreorder]
  int    mTurnsToBuild = 0;
                         ^
src/Things/Structures/../../ProductionCost.h:40:25: warning:   ‘int ProductionCost::mCommonMetals’ [-Wreorder]
  int    mCommonMetals = 0;
                         ^
src/Things/Structures/../../ProductionCost.h:14:2: warning:   when initialized here [-Wreorder]
  ProductionCost(int turns, int commonMetals, int commonMinerals, int rareMetals
  ^
```
Note: Constructor/destruction order is controlled by order of member variable declarations in the class, not the order of the member initializer list. This is to keep consistent construction/destruction order between different constructors and the destructor. Keeping these consistent makes it easier for a compiler to implement sane and safe behaviour if a constructor ever throws an exception.

--- a/src/Map/TileMap.cpp
+++ b/src/Map/TileMap.cpp
@@ -5,6 +5,7 @@
 
 #include "../Constants.h"
 
+#include <algorithm>
 #include <functional>
 #include <random>
 

--- a/src/ProductionCost.h
+++ b/src/ProductionCost.h
@@ -41,5 +41,5 @@ private:
 	int				mCommonMetals = 0;
 	int				mCommonMinerals = 0;
 	int				mRareMetals = 0;
-	int 			mRareMinerals = 0;
+	int				mRareMinerals = 0;
 };

--- a/src/ProductionCost.h
+++ b/src/ProductionCost.h
@@ -37,9 +37,9 @@ public:
 	int turnsToBuild() const { return mTurnsToBuild; }
 
 private:
+	int				mTurnsToBuild = 0;
 	int				mCommonMetals = 0;
 	int				mCommonMinerals = 0;
 	int				mRareMetals = 0;
 	int 			mRareMinerals = 0;
-	int				mTurnsToBuild = 0;
 };

--- a/src/ProductionCost.h
+++ b/src/ProductionCost.h
@@ -23,18 +23,18 @@ public:
 
 	void clear()
 	{
+		mTurnsToBuild = 0;
 		mCommonMetals = 0;
 		mCommonMinerals = 0;
 		mRareMetals = 0;
 		mRareMinerals = 0;
-		mTurnsToBuild = 0;
 	}
 
+	int turnsToBuild() const { return mTurnsToBuild; }
 	int commonMetals() const { return mCommonMetals; }
 	int commonMinerals() const { return mCommonMinerals; }
 	int rareMetals() const { return mRareMetals; }
 	int rareMinerals() const { return mRareMinerals; }
-	int turnsToBuild() const { return mTurnsToBuild; }
 
 private:
 	int				mTurnsToBuild = 0;

--- a/src/RobotPool.cpp
+++ b/src/RobotPool.cpp
@@ -4,6 +4,8 @@
 #include "RobotPool.h"
 #include "RobotPoolHelper.h"
 
+#include <algorithm>
+
 extern int ROBOT_ID_COUNTER; /// \fixme	Kludge
 
 

--- a/src/States/GameState.h
+++ b/src/States/GameState.h
@@ -112,7 +112,7 @@ private:
 	void checkConnectedness();
 	bool changeDepth(int _d);
 
-	void GameState::pullRobotFromFactory(ProductType pt, Factory& factory);
+	void pullRobotFromFactory(ProductType pt, Factory& factory);
 	void factoryProductionComplete(Factory& factory);
 
 	// TURN LOGIC

--- a/src/Things/Structures/RobotCommand.cpp
+++ b/src/Things/Structures/RobotCommand.cpp
@@ -4,6 +4,8 @@
 
 #include "../../Constants.h"
 
+#include <algorithm>
+
 
 /**
  * Gets whether the command facility has additional command capacity remaining.

